app/Helpers/database/game.php

```diff
@@ -24,17 +24,11 @@ function getGameData(int $gameID): ?array
 }
 
 // If the game is a subset, identify its parent game ID.
-function getParentGameIdFromGameTitle(string $title, int $consoleID): ?int
+function getParentGameIdFromGameTitle(string $title, int $consoleId): ?int
 {
-    if (preg_match('/(.+)(\[Subset - .+\])/', $title, $matches)) {
-        $baseSetTitle = trim($matches[1]);
-        $query = "SELECT ID FROM GameData WHERE Title = :title AND ConsoleID = :consoleId";
-        $result = legacyDbFetch($query, ['title' => $baseSetTitle, 'consoleId' => $consoleID]);
+    $foundGame = Game::where('Title', $title)->where('ConsoleID', $consoleId)->first();
```

Comment:
This adds an extra query to every place that was previously calling this function. The "[Subset -" guard should be re-added.

Reply:
Subset guard re-added in latest

Reply:
Adding the subset guard around $foundGame and it's return causes UploadAchievementTest to fail with this error: 
```
Call to a member function getParentGameId() on null
--- Expected
+++ Actual
@@ @@
 '{\n
-    "AchievementID": 2,\n
-    "Error": "",\n
-    "Success": true\n
+    "Error": "Call to a member function getParentGameId() on null",\n
+    "Status": 500,\n
+    "Success": false\n
 }'
```

What else do I need to do here to make sure it returns the correct data?

Reply:
Without seeing your changes, I can't say for certain, but the error implies that you're still calling `$foundGame->getParentGameId()` and `$foundGame` is null. You could change the `->` to `?->`, but it would be simpler to just wrap the logic in a condtional:
```
    if (strstr('[Subset - ', $title) !== false) {
        $foundGame = Game::where('Title', $title)->where('ConsoleID', $consoleId)->first();
        return $foundGame->getParentGameId() ?? null;
    }

    return null;
```

Reply:
This suggestion has been applied to latest.